src/main/java/com/booksaw/betterTeams/team/TeamPlayerSetComponent.java

```diff
@@ -8,93 +8,76 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public abstract class TeamPlayerSetComponent extends SetTeamComponent<TeamPlayer> {
 
 	/**
 	 * @return A list of players which are currently online and on this team
 	 */
 	public List<Player> getOnlinePlayers() {
-		List<Player> online = new ArrayList<>();
-
-		for (TeamPlayer player : getClone()) {
-			if (player.getPlayer().isOnline()) {
-				online.add(player.getPlayer().getPlayer());
-			}
-		}
-
-		return online;
+		return getClone().stream()
+			.map(TeamPlayer::getPlayer)
+			.filter(OfflinePlayer::isOnline)
+			.map(OfflinePlayer::getPlayer)
+			.collect(Collectors.toList());
 	}
 
 	/**
 	 * @return A list of players which are currently online on this team
 	 */
 	public List<OfflinePlayer> getOfflinePlayers() {
-		List<OfflinePlayer> offline = new ArrayList<>();
-
-		for (TeamPlayer player : getClone()) {
-			if (!player.getPlayer().isOnline()) {
-				offline.add(player.getPlayer());
-			}
-		}
-
-		return offline;
+		return getClone().stream()
+			.map(TeamPlayer::getPlayer)
+			.filter(OfflinePlayer::isOnline)
```

Comment:
does this not filter to online players still like the `getOnlinePlayers` method?

Reply:
Thanks a lot for verifying this! That's exactly why PR reviews are needed.
I fixed this.